File: Workshop1/SearchAlgorithms/JumpSearch.py
import math

def jump_search(arr, key):
    n = len(arr)
    step = int(math.sqrt(n))
    prev = 0
    while arr[min(step, n) -1] < key:
        prev = step
        step += int(math.sqrt(n))
        if prev >= n:
            return -1
    for i in range(prev, min(step, n)):
        if arr[i] == key:
            return i
    return -1

File: Workshop1/SearchAlgorithms/JumpSearch.c
int JumpSearch(int arr[], int n, int key){
    int step = (int)sqrt(n);
    int prev = 0;

    while(arr[(step < n ? step : n) - 1]< key)
    {
        prev = step;
        step += (int)sqrt(n);
        if (prev >= n)
        {
            return -1;
        }
    }
    int end = (step < n ? step : n);
    for(int i = prev; i < end; i++)
    {
        if (arr[i] == key)
        {
            return i;
        }
        return -1
    }
}